### ISSUE TITLE
Ordering by a 2nd property appears to be broken in 6.x

### ISSUE THREAD

**Issue by glen-84** [ravendb/ravendb#18708]
```cs
var result = session.Query<Bar>()
    .OrderByDescending(b => b.Foo.BarBool)
    .ThenByDescending(b => b.Foo.BarShort)
    .ToList();
```

With `ravendb/ravendb:5.4-ubuntu-latest`:

```
Foo.BarBool: True, Foo.BarShort: 14
Foo.BarBool: True, Foo.BarShort: 12
Foo.BarBool: False, Foo.BarShort: 13
```

With `ravendb/ravendb:6.0-ubuntu-latest`:

```
Foo.BarBool: True, Foo.BarShort: 12
Foo.BarBool: True, Foo.BarShort: 14     <-- Not descending
Foo.BarBool: False, Foo.BarShort: 13
```

https://github.com/glen-84/ravendb-order-issue

**Comment by Lwiel** [comment]
It's a bug, I've created an issue for this - https://issues.hibernatingrhinos.com/issue/RavenDB-22516/Ordering-by-second-property-doesnt-work-properly-in-Corax.

As a workaround you can switch the search engine to Lucene.

**Comment by arekpalinski** [comment]
It got fixed in https://github.com/ravendb/ravendb/pull/18763. The fix will be available in the next release of RavenDB 6.0.

**Comment by glen-84** [comment]
@Lwiel @arekpalinski

This seems even more broken in `6.0.105`. I'm seeing:

```
Foo.BarBool: False, Foo.BarShort: 13
Foo.BarBool: True, Foo.BarShort: 14
Foo.BarBool: True, Foo.BarShort: 12
```

**Edit:** I've opened #18912.
